### PR TITLE
don't emit invalid turns when having u-turns on ferries in harbours/dead ends

### PR DESCRIPTION
**Pull request by MoKob**
# Issue

resolves https://github.com/Project-OSRM/osrm-backend/issues/3609

## Tasklist
 - [x] add regression / cucumber cases (see docs/testing.md)
 - [x] review
 - [x] adjust for comments

## Requirements / Relations
 none

### CODE REVIEW

**Comment by MoKob** [comment]
For explanation: this is technically broken data, since we require a ferry into nothingness to allow the u-turn in the first place. So we actually don't care what comes out, but we shouldn't fail hard.

**Comment by daniel-j-h** [comment]
The added test is currently failing

https://travis-ci.org/Project-OSRM/osrm-backend/jobs/194849294#L4758

```cucumber
| (-) a,b,1 | (-) land,sea,sea,land,land | (-) depart,notification straight,arrive,depart,arrive | 
| (+) a,b,1 | (+) land,land,sea,land,land | (+) depart,arrive,depart,notification straight,arrive | 
```

except for the mason build where all tests pass happily oO
https://travis-ci.org/Project-OSRM/osrm-backend/jobs/194849291#L4409

**Comment by MoKob** [comment]
@daniel-j-h thats the beauty about via-routes, if one spot is passed twice, you will have to equally valid versions...

File: features/guidance/collapse-ferry.feature [hunk 1]
```diff
@@ -141,3 +141,22 @@ Feature: Collapse
...
+            """
+
+        And the ways
+            | nodes | highway | route | name |
+            | ab    | primary |       | land |
+            | bc    |         | ferry | sea  |
+
+        # we actually cannot check the route here, since two possible routes are equally valid:
+        # (ab)(bcb1) and (abcb)(b1) are exactly the same. Luckily, we only want to check for
+        # not asserting here.
```
Comment (daniel-j-h) [inline features/guidance/collapse-ferry.feature]: :+1:

**Comment by daniel-j-h** [comment]
Ready to go @MoKob :+1: Thanks for the explanation.

**Comment by daniel-j-h** [comment]
Ah and can you backport into 5.5 for a patch release

**Comment by MoKob** [comment]
Seems its not in 5.5, bug came in later. Removing milestone

**Pull request merged**